src/IW/Effects/Download.hs

```diff
@@ -1,36 +1,41 @@
-{- | This module contains the class definitiion of @MonadDownload@ and 
-an instance of @MonadDownload@ for the @App@ monad. Instances of 
+{- | This module contains the class definitiion of @MonadDownload@ and
+an instance of @MonadDownload@ for the @App@ monad. Instances of
 @MonadDownload@ have a @downloadFile@ action for downloading files from a given URL.
 -}
 
 module IW.Effects.Download
        ( MonadDownload (..)
-        
+
        -- * Internals
        , downloadFileImpl
        ) where
 
 import Network.HTTP.Client (Manager, Response (..), httpLbs)
 import Network.HTTP.Types (Status (..))
 
-import IW.App (App, Has, WithError, grab, throwError, notFound)   
-import IW.Core.Url (Url (..)) 
+import IW.App (App, Has, WithError, grab, throwError, notFound)
+import IW.Core.Url (Url (..))
 
 
--- | Describes a monad that can download files from a given URL.
+-- | Describes a monad that can download files from a given @Url@.
 class Monad m => MonadDownload m where
     downloadFile :: Url -> m ByteString
 
-instance MonadDownload App where 
+instance MonadDownload App where
     downloadFile = downloadFileImpl
 
-type WithDownload env m = (MonadIO m, MonadReader env m, WithError m, Has Manager env)
+type WithDownload env m = (MonadIO m, MonadReader env m, WithError m, WithLog env m, Has Manager env)
 
 downloadFileImpl :: WithDownload env m => Url -> m ByteString
 downloadFileImpl Url{..} = do
     man <- grab @Manager
     let req = fromString $ toString unUrl
+    log I $ "Attempting to download file from " <> unUrl <> "..."
     response <- liftIO $ httpLbs req man
     case statusCode $ responseStatus response of
-        200 -> pure $ toStrict $ responseBody response
-        _   -> throwError notFound
+        200 -> do
+            log I $ "Successfully downloaded file from " <> unUrl
+            pure $ toStrict $ responseBody response
+        _   -> do
+            log E $ "Couldn't download file from " <> unUrl
```

Comment:
You can also add to logging message the resulting status code that you received. So if we know, whether it's 404 and there's no such file or 500 (some problems with GitHub)